src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -74,31 +74,33 @@ void CalorimeterHitDigi::init() {
     tRes       = m_cfg.tRes / dd4hep::ns;
     stepTDC    = dd4hep::ns / m_cfg.resolutionTDC;
 
+    // sanity checks
+    if (!m_geo.detector()) {
+        error("Unable to locate geometry");
+        throw std::runtime_error("Unable to locate Geometry Service");
+    }
```

Comment:
I know we probably shouldn't rely on the interface remaining constant in this regards, but this is currently defined as:
```
gsl::not_null<const dd4hep::Detector*> detector() const { return m_detector; }
```
so this cannot ever happen.